### PR TITLE
validate ProjectSpace names

### PR DESCRIPTION
**Pull request by wow-such-code**
Needed to fix https://github.com/qbicsoftware/offer-manager-2-portlet/issues/695

### CODE REVIEW

**Comment by wow-such-code** [comment]
[#695 Fix: check for umlauts in space name before project creation](https://trello.com/c/94Z0gEm9/906-695-fix-check-for-umlauts-in-space-name-before-project-creation)

File: src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy [hunk 1]
```diff
@@ -23,11 +23,21 @@ final class ProjectSpace {
      */
     final String name
 
+    private static final def REGEX = ~'[-\\w]+'
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy:26]: I don't understand this regular expression. What does it match?

Comment (sven1103) [reply src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy:26]: Maybe a more declarative variable name would improve readability as well

**Review by sven1103 — CHANGES_REQUESTED** [review]
Some minor comments only, else good!

File: src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy (hunk 1 above)
Comment (wow-such-code) [reply src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy:26]: + matches the previous token between one and unlimited times, as many times as possible, giving back as needed (greedy)
- matches the character - with index 4510 (2D16 or 558) literally (case sensitive)
\w matches any word character (equivalent to [a-zA-Z0-9_])

not sure about the tilde, but that I took from the project code implementation

Comment (wow-such-code) [reply src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy]: pushed changes

File: src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy [hunk 2]
```diff
@@ -23,7 +23,10 @@ final class ProjectSpace {
      */
     final String name
 
-    private static final def REGEX = ~'[-\\w]+'
+    /**
+     * Alphanumerical characters, underscores and '-' allowed
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectSpace.groovy:27]:
```suggestion
     * Alphanumerical characters, underscores and '-' allowed, German umlauts will lead to a mismatch
```

**Review by sven1103 — APPROVED** [review]
Nice!

**Pull request merged**